spotbugs/src/main/java/edu/umd/cs/findbugs/SourceLineAnnotation.java

```diff
@@ -944,6 +944,34 @@ public String getSourcePath() {
         return sourcePath;
     }
 
+    public String getRealSourcePath() {
+        if (isSourceFileKnown()) {
+            SourceFinder sourceFinder = getSourceFinder();
+            if (sourceFinder != null)
+            {
+                try {
+                    return new File(sourceFinder.findSourceFile(this).getFullFileName()).getCanonicalPath();
+                } catch (IOException e) {
+                    AnalysisContext.logError("Error resolving Real SourcePath (only relative source path will be available) ", e);
+                }
+            }
+            AnalysisContext.logError("No SourceFinder found (only relative source path will be available) ");
```

Comment:
If a SourceFinder exists, but gets an error, the logic will fall through and still print this message.

It would be better to structure it as

    if (sourceFinder == null) {
        AnalysisContext.logError("No SourceFinder found (only relative source path will be available) ");
    } else {
        ...